src/main.js

```diff
@@ -69,6 +69,8 @@ async function initialize() {
   window.gToolbox = new DevToolsToolbox();
   store = await bootstrapStore();
 
+  window.addEventListener("beforeunload", socketWillClose);
```

Comment:
nit: i'd probably just put the listener is socket.js and have several listeners if we need it... not a big deal though

Reply:
Good idea.